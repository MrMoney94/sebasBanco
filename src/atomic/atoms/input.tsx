import React from 'react';
import {StyleSheet, View, TextInput, TextInputProps} from 'react-native';

export const Input = React.forwardRef<TextInput, TextInputProps>(
  (props, ref) => {
    return (
      <View>
        <View style={styles.container}>
          <TextInput ref={ref} {...props} />
        </View>
      </View>
    );
  },
);

const styles = StyleSheet.create({
  container: {
    marginVertical: 5,
    borderRadius: 15,
  },
});
