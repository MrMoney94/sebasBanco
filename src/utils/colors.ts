export const Colors = {
  primary: '#0f265c',
  secondary: '#ffdd00',
  soft_dark: '#ccc',
};
