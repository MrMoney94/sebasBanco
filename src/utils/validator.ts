export const validatorID = (value: string): boolean => {
  return value.length >= 3 && value.length <= 10 && value.trim() !== '';
};

export const validatorName = (value: string): boolean => {
  return value.length >= 5 && value.length <= 100 && value.trim() !== '';
};

export const validatorDescription = (value: string): boolean => {
  return value.length >= 10 && value.length <= 200 && value.trim() !== '';
};

export const validatorLogo = (value: string): boolean => {
  return value.trim() !== '';
};

export const formatDate = (text: string): string => {
  const formattedText = text
    .replace(/\D/g, '')
    .replace(/(\d{2})(\d{2})(\d{4})/, '$1/$2/$3');
  return formattedText;
};

export const isValidDate = (inputDate: string): boolean => {
  const currentDate = new Date();
  const selectedDate = new Date(
    inputDate.replace(/\//g, '-').split('-').reverse().join('-'),
  );
  const compare = selectedDate.toISOString().split('T')[0];

  return compare >= currentDate.toISOString().split('T')[0];
};
