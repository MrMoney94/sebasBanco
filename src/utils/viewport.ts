import {Dimensions, PixelRatio} from 'react-native';

const {width, height} = Dimensions.get('window');

export const widthSize = width;

export const heightSize = height;

export const FontSize = (size: number) => {
  return PixelRatio.getFontScale() * size;
};
