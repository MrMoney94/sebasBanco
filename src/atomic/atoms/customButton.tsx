import React from 'react';
import {
  StyleSheet,
  Text,
  Pressable,
  PressableProps,
  ActivityIndicator,
} from 'react-native';
import {widthSize} from '../../utils/viewport';
import {Colors} from '../../utils/colors';

type Props = PressableProps & {
  title: string;
  loading?: boolean;
  color?: string;
};

export const CustomButton: React.FC<Props> = props => {
  return !props.loading ? (
    <Pressable
      style={[
        styles.buttonStyle,
        {backgroundColor: props.color ?? Colors.secondary},
      ]}
      {...props}>
      <Text style={styles.font}>{props.title}</Text>
    </Pressable>
  ) : (
    <ActivityIndicator size="small" color={Colors.primary} />
  );
};

const styles = StyleSheet.create({
  buttonStyle: {
    paddingVertical: 12,
    width: widthSize / 1.2,
    borderRadius: 10,
    marginVertical: 5,
  },
  font: {
    color: Colors.primary,
    textAlign: 'center',
    fontSize: 16,
    fontWeight: '600',
    paddingVertical: 6,
  },
});
