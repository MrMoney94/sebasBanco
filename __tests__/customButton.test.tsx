import React from 'react';
import renderer from 'react-test-renderer';
import {Pressable} from 'react-native';
import {CustomButton} from '../src/atomic/atoms/customButton';
import {Colors} from '../src/utils/colors';

describe('CustomButton', () => {
  it('renders correctly when not loading', () => {
    const tree = renderer.create(<CustomButton title="Press me" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when loading', () => {
    const tree = renderer
      .create(<CustomButton title="Press me" loading={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies the correct background color', () => {
    const tree = renderer
      .create(<CustomButton title="Press me" color={Colors.primary} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('handles press events', () => {
    const onPressMock = jest.fn();
    const tree = renderer.create(
      <CustomButton title="Press me" onPress={onPressMock} />,
    );
    // Usa el componente importado Pressable
    tree.root.findByType(Pressable).props.onPress();
    expect(onPressMock).toHaveBeenCalled();
  });
});
