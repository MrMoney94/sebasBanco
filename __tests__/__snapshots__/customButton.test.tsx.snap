// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton applies the correct background color 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  color="#0f265c"
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "borderRadius": 10,
        "marginVertical": 5,
        "paddingVertical": 12,
        "width": 625,
      },
      {
        "backgroundColor": "#0f265c",
      },
    ]
  }
  title="Press me"
>
  <Text
    style={
      {
        "color": "#0f265c",
        "fontSize": 16,
        "fontWeight": "600",
        "paddingVertical": 6,
        "textAlign": "center",
      }
    }
  >
    Press me
  </Text>
</View>
`;

exports[`CustomButton renders correctly when loading 1`] = `
<ActivityIndicator
  color="#0f265c"
  size="small"
/>
`;

exports[`CustomButton renders correctly when not loading 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "borderRadius": 10,
        "marginVertical": 5,
        "paddingVertical": 12,
        "width": 625,
      },
      {
        "backgroundColor": "#ffdd00",
      },
    ]
  }
  title="Press me"
>
  <Text
    style={
      {
        "color": "#0f265c",
        "fontSize": 16,
        "fontWeight": "600",
        "paddingVertical": 6,
        "textAlign": "center",
      }
    }
  >
    Press me
  </Text>
</View>
`;
